### PR TITLE
TIG-1397: Crud Actor 💅

### PR DESCRIPTION
**Pull request by rtimmons**
Few light housekeeping things before I add more functionality:

1. switch to `BOOST_THROW_EXCEPTION`
2. `std::move` in operation ctor params
3. Record documents and bytes metrics where appropriate

Sys-perf patch: https://evergreen.mongodb.com/version/5cb0d1c50ae606414132d48b

### CODE REVIEW

File: src/cast_core/src/CrudActor.cpp [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#include <utility>
```
Comment (visemet) [inline src/cast_core/src/CrudActor.cpp]: It looks like this include got added to the wrong place.

**Comment by rtimmons** [comment]
Thanks for review.

> Should we call ctx.addDocuments() with the return value of _collection.count_documents()?

Was this in reference to a particular line? GH didn't attach it anywhere....

**Comment by visemet** [comment]
> > Should we call ctx.addDocuments() with the return value of _collection.count_documents()?
> 
> Was this in reference to a particular line? GH didn't attach it anywhere....

It's in reference to `CountDocumentsOperation::run()`. GitHub only allows you to comment on a few lines surrounding the diff. You didn't change it so I couldn't comment on it 🙃

**Comment by rtimmons** [comment]
Updated per feedback - ready for 👀 when you can

**Pull request merged**